src/Rules/Deprecations/DeprecatedHookImplementation.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace mglaman\PHPStanDrupal\Rules\Deprecations;
+
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Function_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+
+class DeprecatedHookImplementation implements Rule
+{
+
+    /**
+     * @var \PHPStan\Reflection\ReflectionProvider
+     */
+    protected $reflectionProvider;
+
+    public function __construct(ReflectionProvider $reflectionProvider)
+    {
+        $this->reflectionProvider = $reflectionProvider;
+    }
+
+    public function getNodeType(): string
+    {
+        return Function_::class;
+    }
+
+    public function processNode(Node $node, Scope $scope) : array
+    {
+        assert($node instanceof Function_);
+        if (!str_ends_with($scope->getFile(), ".module") && !str_ends_with($scope->getFile(), ".inc")) {
```

Comment:
phpstan-drupal still supports PHP 7.4 technically for Drupal 9. But we can trust the PHP 8 polyfill exists for D9, so it should be okay right?

Reply:
Also, Drupal 9 officially doesn't support 7.4 anymore ;)

Reply:
"officially"

Reply:
Probably no breaking code, haven't noticed, just bo promised I think

Reply:
I've used `str_ends_with` and `str_starts_with` for pretty much the entirety of Drupal 9's lifetime :D The `symfony/polyfill-php80` package takes care of the functions (and Drupal depends on it).

Not sure if we need to add a dependency on that package explicitly in phpstan-drupal?

Reply:
Nah, I think we're OK and I should probably just drop PHP 7.4. I was waiting for PHPStan to drop it, but the Drupal community has pretty much moved onto PHP 8.1 (it seems.)

Nothing to do here, though.